src/icrar/leap-accelerate/ms/MeasurementSet.cc

```diff
@@ -153,6 +155,35 @@ namespace icrar
         }
     }
 
+    Eigen::Matrix<bool, -1, 1> MeasurementSet::GetFlaggedBaselines() const
+    {
+        // TODO: may want to consider using logical OR over for each channel and polarization.
+        auto epochIndices = casacore::Slice(0, GetNumBaselines(), 1);
+        auto nBaselines = GetNumBaselines();
+        auto flagSlice = casacore::Slicer(
+            casacore::IPosition(3,0,0,0),
+            casacore::IPosition(3,1,1,nBaselines),
+            casacore::IPosition(3,1,1,1));
+        casacore::Vector<bool> fg = m_msmc->flag().getColumn()
+        (flagSlice).reform(casacore::IPosition(1, nBaselines))
+        (epochIndices);
```

Comment:
If you are only accessing a slice within each cell, and only certain number of rows, you're better off using `getColumnRange` (which will read only what's needed) instead of `getColumn` + subindexing (which will use more memory). Something like this:

```c++
    Eigen::Matrix<bool, -1, 1> MeasurementSet::GetFlaggedBaselines() const
    {
        // TODO: may want to consider using logical OR over for each channel and polarization.
        auto nBaselines = GetNumBaselines();
        auto rowRange = casacore::Slicer(casacore::Slice(0, nBaselines));
        auto flagSlice = casacore::Slicer(
            casacore::IPosition(2, 0, 0),
            casacore::IPosition(2, 1, 1),
            casacore::IPosition(2, 1, 1));
        casacore::Vector<bool> flags = m_msmc->flag().getColumnRange(rowRange, flagSlice);
        return ToVector(flags);
    }
```

There are other places in the code where the same happens (i.e., a full column is read, then indexed to extract a part of its contents), so little by little we should move towards using the more memory-efficient methods.